contracts/OtokenFactory.sol

```diff
@@ -0,0 +1,177 @@
+pragma solidity =0.6.10;
+
+import {Spawner} from "./packages/Spawner.sol";
+import {IAddressBook} from "./interfaces/IAddressBook.sol";
+import {IWhitelistModule} from "./interfaces/IWhitelistModule.sol";
+
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ * @title A factory for opyn tokens
+ * @author Opyn
+ * @notice Create new otokens and keep track of all created tokens.
+ * @dev Calculate contract address before each creation with CREATE2
+ * and deploy eip-1167 minimal proxies for otoken logic contract.
+ */
+contract OtokenFactory is Spawner {
+    // The Opyn AddressBook contract that records addresses of whitelist module and oToken impl contract.
+    IAddressBook public addressBook;
+
+    address[] private _otokens;
+
+    mapping(bytes32 => address) private _tokenAddresses;
+
+    constructor(IAddressBook _addressBook) public {
+        addressBook = _addressBook;
+    }
+
+    event OtokenCreated(
+        address tokenAddress,
+        address creator,
```

Comment:
can we add `indexed` here ?

Reply:
Updated. we should add event definition to our master doc as well.